.github/workflows/release.yml

```diff
@@ -3,60 +3,109 @@ name: Release
 on:
   workflow_dispatch:
     inputs:
-      deploy-to:
-        type: choice
-        description: Choose where to publish (test/prod)
-        options:
-        - prod
-        - test
-        default: prod
-      ref:
-        description: "The ref (sha or branch name) to use"
+      branch:
+        description: "The branch to release from"
         type: string
         default: "main"
+      version:
+        description: "The version to release"
         required: true
+        type: string
+      deploy-to:
+        description: "Deploy to test or prod"
+        type: environment
+        default: prod
+      only_docker:
+        description: "Only release Docker image, skip GitHub & PyPI"
+        type: boolean
+        default: false
```

Comment:
Does this imply you can release GitHub & PyPI & Docker or GitHub & PyPI but not Docker by itself?

Reply:
We added this primarily for testing if I recall. PyPI is independent of the other releases. GitHub can be released on its own. Docker requires a GitHub release because it points to the tag. This means it requires a full release as a test release does not create a tag, just a draft release. There is the scenario where we previously released PyPI and GitHub, but not Docker (e.g. because it was not automated at that point). In that case, we want to run only Docker as I believe the others fail due to the existing release. We should really rework this to skip existing releases with a warning instead of failing, but that's better done with the refreshed release pipeline work that will accompany the `pyproject.toml`/`hatch` migration.